doc/rtd/topics/datasources/vmware.rst

```diff
@@ -0,0 +1,359 @@
+keys) cleared as soon as its data is read by the datasource. This is possible
+by adding the following to the metadata:
+
+.. code-block:: yaml
+
+   cleanup-guestinfo:
+   - userdata
+   - vendordata
+
+When the above snippet is added to the metadata, the datasource will iterate
+over the elements in the ``cleanup-guestinfo`` array and clear each of the
+keys. For example, the above snippet will cause the following commands to be
+executed:
+
+.. code-block:: shell
+
+   vmware-rpctool "info-set guestinfo.userdata ---"
+   vmware-rpctool "info-set guestinfo.userdata.encoding  "
+   vmware-rpctool "info-set guestinfo.vendordata ---"
+   vmware-rpctool "info-set guestinfo.vendordata.encoding  "
+
+Please note that keys are set to the valid YAML string ``---`` as it is not
+possible remove an existing key from the guestinfo key-space. A key's analogous
+encoding property will be set to a single white-space character, causing the
+datasource to treat the actual key value as plain-text, thereby loading it as
+an empty YAML doc (hence the aforementioned ``---``\ ).
+
+Reading the local IP addresses
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This datasource automatically discovers the local IPv4 and IPv6 addresses for
+a guest operating system based on the default routes. However, when inspecting
+a VM externally, it's not possible to know what the *default* IP address is for
+the guest OS. That's why this datasource sets the discovered, local IPv4 and
+IPv6 addresses back in the guestinfo namespace as the following keys:
+
+
+* ``guestinfo.local-ipv4``
+* ``guestinfo.local-ipv6``
+
+It is possible that a host may not have any default, local IP addresses. It's
+also possible the reported, local addresses are link-local addresses. But these
+two keys may be used to discover what this datasource determined were the local
+IPv4 and IPv6 addresses for a host.
+
+Waiting on the network
+^^^^^^^^^^^^^^^^^^^^^^
+
+Sometimes cloud-init may bring up the network, but it will not finish coming
+online before the datasource's ``setup`` function is called, resulting in an
+``/var/run/cloud-init/instance-data.json`` file that does not have the correct
+network information. It is possible to instruct the datasource to wait until an
+IPv4 or IPv6 address is available before writing the instance data with the
+following metadata properties:
+
+.. code-block:: yaml
+
+   wait-on-network:
+     ipv4: true
+     ipv6: true
+
+If either of the above values are true, then the datasource will sleep for a
+second, check the network status, and repeat until one or both addresses from
+the specified families are available.
+
+Walkthrough
+-----------
+
+The following series of steps is a demonstration on how to configure a VM with
+this datasource:
+
+
+#. Create the metadata file for the VM. Save the following YAML to a file named
+   ``metadata.yaml``\ :
+
+   .. code-block:: yaml
+
+       instance-id: cloud-vm
+       local-hostname: cloud-vm
+       network:
+       version: 2
```

Comment:
Perhaps. It works as-is though (I tested it by running `make doc` and opening `index.html` locally). The problem is the `m2r` command does not convert the original markdown very well at all, and it does not offer a way to specify tab width. I am new to `rst`, and the other datasource docs use varying widths, two, three, four. It was unclear if there is a standard width (`m2r` chose three), and since the original markdown used four, I just verified things looked correct in the browser and was going to ask about indentation policy later (again, there doesn't seem to be a standard based on the docs for the other datasources).

Reply:
Sorry, for this I don't mean the rst indentation, I mean that the section that comes under `network` should be further indented as part of the network definition in the yaml.

Reply:
Ugh, yes. VSCode and it's block tab indentation. Usually I have to just do a find/replace with a regex `s/^/<WHITE_SPACE>/g` to achieve consistent, leading white space. I guess I did not do that here.